daml/splice-amulet-test/daml/Splice/Scripts/TestValidatorFaucet.daml

```diff
@@ -102,3 +106,160 @@ test_ValidatorFaucet = do
+      openRoundCid = round0Cid
+    let aliceLicenseCid = aliceResult.licenseCid
+
+    bobResult <- submitMulti [bobValidator.primaryParty] [app.dso] $ exerciseCmd bobLicenseCid ValidatorLicense_RecordValidatorLivenessActivity with
+      openRoundCid = round0Cid
+    let bobLicenseCid = bobResult.licenseCid
+
+    -- Verify the activity records were created with correct weights
+    aliceRecords <- queryFilter @ValidatorLivenessActivityRecord app.dso
+      (\r -> r.validator == aliceValidator.primaryParty && r.round == round0.round)
+    bobRecords <- queryFilter @ValidatorLivenessActivityRecord app.dso
+      (\r -> r.validator == bobValidator.primaryParty && r.round == round0.round)
+
+    length aliceRecords === 1
+    length bobRecords === 1
+
+    let [(_, aliceRecord)] = aliceRecords
+    let [(_, bobRecord)] = bobRecords
+
+    aliceRecord.weight === None
+    bobRecord.weight === Some bobWeight
+
+    runNextIssuance app
+    runNextIssuance app
+
+    runAmuletDepositBots app
+
+    aliceBalanceAfter <- getNormalizedBalance aliceValidator.primaryParty
+    bobBalanceAfter <- getNormalizedBalance bobValidator.primaryParty
+
+    let aliceReward = aliceBalanceAfter - aliceBalanceBefore
+    let bobReward = bobBalanceAfter - bobBalanceBefore
+
+    let aliceExpectedMin = 2.8
+    let aliceExpectedMax = 3.0
+
+    require "Alice should receive default rewards" (aliceReward >= aliceExpectedMin && aliceReward <= aliceExpectedMax)
+
+    case bobExpectedMin of
+      Some minVal -> require "Bob should receive expected reward based on weight" (bobReward >= minVal && bobReward <= bobExpectedMax)
+      None -> require "Bob should not receive rewards" (bobReward <= bobExpectedMax)
+
+    return ()
+
+-- With weight 3.0, Bob should receive 3x Alice's rewards
+test_ValidatorFaucetWithWeight3: Script ()
+test_ValidatorFaucetWithWeight3 = do
+    let bobWeight = 3.0
+        bobExpectedMin = Some 8.4
+        bobExpectedMax = 9.0
+    testValidatorFaucetWithWeight_Helper bobWeight bobExpectedMin bobExpectedMax
+
+-- With weight 0.0, Bob should receive no rewards
+test_ValidatorFaucetWithWeight0: Script ()
+test_ValidatorFaucetWithWeight0 = do
+    -- bob actually loses balance when doing minting with 0 weight. So just confirm bobExpectedMax
+    let bobWeight = 0.0
+        bobExpectedMin = None
+        bobExpectedMax = 0.0
+    testValidatorFaucetWithWeight_Helper bobWeight bobExpectedMin bobExpectedMax
+
+-- Test that runNextIssuance properly handles ValidatorLicense weights
+-- in the optTotalValidatorLivenessActivityRecords calculation
+-- Bob has been given a large weight to trigger the use of `cappedRewardsToIssue` (in computeIssuanceTranche)
+-- which would result in alice getting lower rewards than the default value of about 2.85
+test_ValidatorLivenessWeightInRunNextIssuance : Script ()
+test_ValidatorLivenessWeightInRunNextIssuance = do
```

Comment:
Why can't you reuse the helper here? The code seems very similar. Just a few params missing.